src/sentry/api/endpoints/project_details.py

```diff
@@ -423,6 +425,28 @@ def validate_sensitiveFields(self, value):
     def validate_safeFields(self, value):
         return validate_pii_selectors(value)
 
+    def validate_targetSampleRate(self, value):
+        from sentry import features
+
+        organization = self.context["project"].organization
+        actor = self.context["request"].user
+        if not features.has("organizations:dynamic-sampling-custom", organization, actor=actor):
+            raise serializers.ValidationError(
+                "Organization does not have the custom dynamic sample rate feature enabled."
+            )
+
+        if organization.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT) != "project":
```

Comment:
nit: Do we want to unite this with [choices](https://github.com/getsentry/sentry/blob/bc8d10ea18505efff605bfae3833ff82b8103809/src/sentry/api/endpoints/organization_details.py#L288) for `samplingMode` from `src/sentry/api/endpoints/organization_details.py` and have a single source of truth which modes are possible?